docs/index.md

```diff
@@ -1,24 +1,32 @@
 # autora-theorist-Avengers
 
-Explain what your contribution is doing here
+## Search Algorithm
 
-You can include inline mathematics by wrapping LaTeX formatted math in `$` symbols: `$(3 < 4)$` will render as:  $(3 < 4)$
+The search algorithm employed by the theorist involves fitting a quadratic model to the provided dataset using the `ParabolaRegression` class. The algorithm follows these steps:
 
-To include block mathematics wrap the expression with double `$$`:
+1. **Data Preparation**: The input data \( X \) is converted to a 2D numpy array.
+2. **Design Matrix Construction**: A design matrix \( A \) is constructed, which includes:
+   - An intercept term (a column of ones).
+   - Linear terms (the original features).
+   - Log-transformed terms (log of the original features).
+   - Quadratic terms (squares and cross-products of the features).
+3. **Coefficient Calculation**: The coefficients are calculated using the least squares method by solving the normal equation \( A \beta = y \).
+4. **Model Fitting**: The model is fitted to the data by storing the calculated coefficients.
 
-```
-$$
-E(\mathbf{v}, \mathbf{h}) = -\sum_{i,j}w_{ij}v_i h_j - \sum_i b_i v_i - \sum_j c_j h_j
-$$
-```
+The goodness of the equation is determined using statistical measures such as the coefficient of determination (R²) and mean squared error (MSE).
```

Comment:
The goodness of the fit of the equation to the data is determined using the following statistical measures which are commonly used in regression:

     -   R² -> The R² value measures the amount of variance present in the values of the dependent variable that can or cannot be properly explained by the independent variable. A R² value of 0 means the independent variables in question cannot explain the variances present in the dependent variable, whereas a R² value of 1 means that the independent variable can fully account for the variances in the dependent variable.
     
     - Mean Squared Error (MSE) -> The MSE value denotes the mean of the squared error difference between the predicted output values or labels and the original output values or labels. In the case of the parabolic function used by our theorist for fitting the data, the MSE is calculated between the actual output data points and the data points present on the parabola curve calculated by our theorist. This MSE value needs to be minimized by finding optimal coefficient values for each of the input feature terms present in the final parabola equation.